docs/data.rst

```diff
@@ -98,32 +100,109 @@ Note that you can get the XML definition of an IDL file with *rtiddsgen -convert
 We will refer to an ``Output`` named ``output`` and
 ``Input`` named ``input`` such that ``input.sample_count > 0``.
 
-Accessing numbers, strings and boolean members
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Using dictionaries vs accessing individual members
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-TODO: complete
+In an Input or an Output you can access the data all at once, using a dictionary,
+or member by member. Using a dictionary is usually more efficient if you intend
+to access most or all of the data members of a large type.
 
-Accessing nested members
-^^^^^^^^^^^^^^^^^^^^^^^^
+In an Output, :meth:`Instance.set_dictionary` receives a dictionary with all or
+some of the Output type members, and in an Input, :meth:`SampleIterator.get_dictionary`
+retrieves all the members.
 
-The "." notation allows accessing a nested field, for example:
+On the other hand the methods described in the following section receive a
+``field_name`` argument to get or set a specific member.
+
+Accessing basic members (numbers, strings and booleans)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To set a field in an :class:`Output`, use the appropriate setter.
+
+To set any numeric type, including enumerations:
+
+.. testcode::
+
+    output.instance.set_number("my_long", 2)
+    output.instance.set_number("my_double, 2.14)
+    output.instance.set_number("my_enum", 2)
+
+To set booleans:
+
+.. testcode::
+
+    output.instance.set_boolean("my_boolean", True)
+
+To set strings:
+
+.. testcode::
+
+    output.instance.set_string("my_string", "Hello, World!")
+
+
+As an alternative to the setters, the special method ``__setitem__`` can be used:
```

Comment:
Although __setitem__ is being used, it might be clearer to people (especially anybody who doesn't isn't familiar with magic methods) to clarify that __setitem__ is how we implement it internally to allow for the output.instance[] syntax.